torch_xla/experimental/xla_sharding.py

```diff
@@ -331,13 +331,44 @@ def _get_sharding_type(partition_spec: Tuple[Union[int, None]],
   return sharding_type
 
 
-def _get_tile_assignment(mesh: Mesh,
-                         partition_spec: Tuple[Union[int, None]]) -> List[int]:
-  if (None not in partition_spec) and (len(mesh.mesh_shape)
-                                       == len(partition_spec)):
-    return mesh.get_logical_mesh().transpose(partition_spec).tolist()
-  # Tile permutation is not necessary for partial replication.
-  return mesh.get_logical_mesh().tolist()
+def _get_tile_assignment(
+    mesh: Mesh, partition_spec: Tuple[Union[Tuple[int], int, None]]
+) -> Tuple[np.ndarray, Tuple[Union[int, None]]]:
+  """
+  Permute the given mesh to create the tile assignment based on the partition
+  spec. Returns the tiling assignment as a numpy ndarray and an updated
+  partition spec as Tuple of int or None.
+
+  If the input partition_spec combines multiple logical mesh axes over a single
+  tensor axis, the returned partition spec will be updated to remove the
+  grouping, and its indices will refer to the updated tile shape. The resulting
+  tiling assignment will combine the specified axes into a single axis.
+
+  The resulting partition spec and tiling assignment should be used to generate
+  the group assignment and replica groups in the case of partial replication,
+  and the tiling assignment should be used to create the final sharding spec.
+  """
+  # Flatten the partition spec and ensure that it is fully specified over the
+  # mesh for permutation.
+  tiled_dims = [x for x in partition_spec if x is not None]
+  permutation = np.hstack(tiled_dims).tolist() if tiled_dims else []
+  missing_axes = sorted(set(range(len(mesh.shape()))) - set(permutation))
+  tile_assignment = mesh.get_logical_mesh().transpose(permutation +
```

Comment:
I was shocked by the transpose grammar and then realized it's np.array. I'm curious what's the advantage to use np.array over cpu torch.tensor. This is just for discussion. There is no needs to make any changes.